src/lib/errors.ts

```diff
@@ -262,20 +262,23 @@ export class ClientError extends BaseError {
   }
 }
 
-export class WalkedTooManyFilesError extends BaseError {
-  isBug = IsBug.NO;
+export class InvalidSyncFileError extends BaseError {
+  isBug = IsBug.MAYBE;
 
-  constructor(readonly dir: string, readonly maxFiles: number) {
-    super("GGT_CLI_TOO_MANY_FILES", "Found too many files while scanning directory");
+  constructor(override readonly cause: unknown, readonly argv: string[]) {
+    super("GGT_CLI_INVALID_SYNC_FILE", "The .ggt/sync.json file was invalid or not found");
   }
 
-  override body(_: Config): string {
+  protected body(config: Config): string {
     return dedent`
-        The following directory has over ${this.maxFiles} non-ignored files inside of it.
-        ${this.dir}
+      We can't determine the state of your local files relative to your remote ones.
```

Comment:
What do you think of something like:

```
We failed to read the Gadget metadata file in this directory `.ggt/sync.json`.

If you are sync'ing for the first time, we recommend using an empty directory such as `~/gadget/${config.app}`.

Otherwise, if you are sure you want to sync the contents of ${config.dir} to Gadget, you can use the `--force` flag:

...
```

Reply:
Does this work?

```
$ ggt sync --app myapp ~/bad-folder
GGT_CLI_INVALID_SYNC_FILE: The .ggt/sync.json file was invalid or not found

We failed to read the Gadget metadata file in this directory:

  /Users/scott/bad-folder

Without this metadata, we can't determine the state of your local files relative to your remote ones.

If you're running `ggt sync` for the first time, we recommend using an empty directory such as `~/gadget/myapp`.

Otherwise, if you're sure you want to sync the contents of `/Users/scott/bad-folder` to Gadget, you can run the command again with the `--force` flag:

  $ ggt sync --app myapp ~/bad-folder --force

You will be prompted to either merge your local files with your remote ones or reset your local files to your remote ones.

If you think this is a bug, please submit an issue using the link below.

https://github.com/gadget-inc/ggt/issues/new?body=%23%23%23+Description%0A%5BPlease+describe+what+you+were+doing+when+this+error+occurred%5D%0A%0A%23%23%23+Command%0A%60%60%60sh-session%0Aggt+sync+--app+scott%0A%60%60%60%0A%0A%23%23%23+Environment%0A%60%60%60%0Aversion%3A+0.0.0-alpha.1%0Aplatform%3A+darwin%0Aarch%3A+arm64%0Ashell%3A+fish%0Atimestamp%3A+2022-08-18T14%3A09%3A14.433Z%0A%60%60%60%0A%0A%23%23%23+Stack+Trace%0A%60%60%60%0AError%3A+The+.ggt%2Fsync.json+file+was+invalid+or+not+found%0A++++at+Sync.init+%28%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2Fsrc%2Fcommands%2Fsync.ts%3A217%3A17%29%0A++++at+async+Sync._run+%28%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2Fnode_modules%2F%40oclif%2Fcore%2Flib%2Fcommand.js%3A72%3A13%29%0A++++at+async+Config.runCommand+%28%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2Fnode_modules%2F%40oclif%2Fcore%2Flib%2Fconfig%2Fconfig.js%3A270%3A25%29%0A++++at+async+Object.run+%28%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2Fnode_modules%2F%40oclif%2Fcore%2Flib%2Fmain.js%3A76%3A5%29%0A%60%60%60%0A%23%23%23+Cause%0A%60%60%60json%0A%7B%0A++%22errno%22%3A+-2%2C%0A++%22code%22%3A+%22ENOENT%22%2C%0A++%22syscall%22%3A+%22open%22%2C%0A++%22path%22%3A+%22%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2F.ggt%2Fsync.json%22%2C%0A++%22name%22%3A+%22Error%22%2C%0A++%22message%22%3A+%22ENOENT%3A+no+such+file+or+directory%2C+open+%27%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2F.ggt%2Fsync.json%27%22%2C%0A++%22stack%22%3A+%22Error%3A+ENOENT%3A+no+such+file+or+directory%2C+open+%27%2FUsers%2Fscott%2FCode%2Fgadget%2Fggt%2F.ggt%2Fsync.json%27%22%0A%7D%0A%60%60%60&title=%5BBUG+GGT_CLI_INVALID_SYNC_FILE%5D%3A+&labels=bug
```

Reply:
> Without this metadata, we can't determine the state of your local files relative to your remote ones.

I don't think we really need to get this across - it feels like we're leaking a lot of the implementation here. Would you be ok with the text as you wrote it except that line?